pkg/services/baremetal/host/host.go

```diff
@@ -588,6 +588,31 @@ func (s *Service) actionRegistering() actionResult {
 		return s.recordActionFailure(infrav1.RegistrationError, err.Error())
 	}
 
+	// Check RAID for the second time.
+	// This happens for new machines which don't have rootDeviceHints.
+	// See "tworaidchecks" for the other place.
+	msg = ""
+	if s.scope.HetznerBareMetalHost.Spec.Status.InstallImage == nil {
```

Comment:
do you have a reason for this check? Because the host takes installimage != nil as a reason to start provisioning. IF this is not set, we would never reach this state. 

Reference: "NeedsProvisioning()" function

Reply:
> do you have a reason for this check? Because the host takes installimage != nil as a reason to start provisioning. IF this is not set, we would never reach this state.
> 
> Reference: "NeedsProvisioning()" function

Some tests failed, and I got null-pointer exceptions.

If you have a better solution, then please tell me.

Reply:
I checked the code again and I see no reason why InstallImage shouldn't be specified at this point. If this is needed, I believe we have a bug somewhere. Is this easily reproducible or happening sometimes?

Reply:
@guettli I believe the reason is that you have added this here to this function and it wasn't there before. This probably fails in the unit tests (I guess you meant that) because we did not "properly" define the host objects with InstallImage specified, as it wasn't necessary before. 

The solution is to define the host objects in the unit tests appropriately to this change in the function.

Reply:
@janiskemper what do you want:

Image for some strange reason in the future the value is nil in production.

What do you want to happen?

option 1: a null pointer exception

option 2: My current code

option 3: We do a panic("Internal Error: HetznerBareMetalHost.Spec.Status.InstallImage is nil. setHostSpec() not called yet?")

... or something else?

Reply:
> The solution is to define the host objects in the unit tests appropriately to this change in the function.

That is already fixed.

Reply:
so you say we don't need this check "installimage != nil" anymore?

If it is still needed, we should rather change the tests instead of changing the functions based on our tests

Reply:
we update the tests so that we don't have to change the actual code to comply to wrong tests! If you don't know how that works, let's talk about it

Reply:
> we update the tests so that we don't have to change the actual code to comply to wrong tests! If you don't know how that works, let's talk about it

reading between the lines, I guess you want to say: "Please choose option 1". Is this correct?

Reply:
the tests should be configured in a way that there is no issue here with the code or nil pointers.. If we have always InstallImage configured for tests of this function, this never occurs.

Reply:
I see that you removed the InstallImage != nil check without changing the unit tests. So there was no problem in the first place? I'm confused. If you wanna explain me why you added it first and why you removed it now without changing anything else, then let's talk about it

Reply:
@janiskemper 

I changed code. Look for `host.Spec.Status.InstallImage = &infrav1.InstallImage{}`

And the second change was that: `helpers.WithRootDeviceHintWWN(),`

Both changes were done before I requested the review from you.